Duplicate song warning added

RitikPandey1 opened the pull request:
Fixes #554 

If user skip :
![navidromeSkip](https://user-images.githubusercontent.com/57366926/111037140-d9b59580-8448-11eb-9286-b639cd736528.gif)

If user want to add song : 
![navidromeAdd](https://user-images.githubusercontent.com/57366926/111037164-fbaf1800-8448-11eb-911f-3aa9e25d18d9.gif)

Tell me if any changes are required @deluan 
Happy to contribute :)

deluan reviewed (changes requested):
Hey @RitikPandey1 , this looks great! A couple of notes, though:
- I think it would be better to leave the "Add" button of the `AddToPlaylist` dialog disabled, and only enable it if the `checkDuplicateSong` determine it is not duplicated, or if the user clicks "Add" in the DuplicateSongDialog
- Ideally we would want to validate if the song is already in the playlist on the server side, but as we are using this issue as a "simple" task to evaluate students for GSoC, a full client-side solution is ok-ish (except when the playlist has thousands of songs, and we will have to retrieve the whole playlist just to check for duplicates.... But this can be tackled later). If you want to try implementing the server side, let me know
- We should extend the check for all songs being added to the playlist, instead of making it work when adding just one song. Maybe this is more complicated for you to tackle. Let me know.

Besides that, the build was not successful, I think because you didn't run `prettier` before commiting. Try configuring your editor to auto run `prettier` on save

Anyway, good job!

deluan commented:
Regarding my first point in the review, you could do something like what Apple Music does: If at least one music is duplicated, it shows the following dialog:
<img width="645" alt="Screen Shot 2021-03-13 at 9 58 50 PM" src="https://user-images.githubusercontent.com/331353/111055798-82fb8c00-8447-11eb-9161-c32ee42ae699.png">

The user then can Skip the duplicated, go ahead and Add everything, or just Cancel the whole operation. It is the same dialog regardless if the user is adding one or multiple songs. That simplifies the implementation, and provides a good, uniform UX (IMHO)

RitikPandey1 commented:
> Hey @RitikPandey1 , this looks great! A couple of notes, though:
> 
> * I think it would be better to leave the "Add" button of the `AddToPlaylist` dialog disabled, and only enable it if the `checkDuplicateSong` determine it is not duplicated, or if the user clicks "Add" in the DuplicateSongDialog
> * Ideally we would want to validate if the song is already in the playlist on the server side, but as we are using this issue as a "simple" task to evaluate students for GSoC, a full client-side solution is ok-ish (except when the playlist has thousands of songs, and we will have to retrieve the whole playlist just to check for duplicates.... But this can be tackled later). If you want to try implementing the server side, let me know
> * We should extend the check for all songs being added to the playlist, instead of making it work when adding just one song. Maybe this is more complicated for you to tackle. Let me know.
> 
> Besides that, the build was not successful, I think because you didn't run `prettier` before commiting. Try configuring your editor to auto run `prettier` on save
> 
> Anyway, good job!

Thank you @deluan for review.
Regarding your second point, I'am totally agree with you. Performing duplicate check on client-side is not efficient, but currently I'am learning `go` and not much familiar with it, Once I'll become familiar to `go` then I would love to implement all these features on backend also. I would also like to contribute to navidrome under GSOC.
I have tried my best to  implemented the features(on client-side)  you suggested in first and third point of your review.

RitikPandey1 commented:
> Regarding my first point in the review, you could do something like what Apple Music does: If at least one music is duplicated, it shows the following dialog:
> <img alt="Screen Shot 2021-03-13 at 9 58 50 PM" width="645" src="https://user-images.githubusercontent.com/331353/111055798-82fb8c00-8447-11eb-9161-c32ee42ae699.png">
> 
> The user then can Skip the duplicated, go ahead and Add everything, or just Cancel the whole operation. It is the same dialog regardless if the user is adding one or multiple songs. That simplifies the implementation, and provides a good, uniform UX (IMHO)

I have added same dialog for single song and multiple songs, this make implementation easy.
Thank you

RitikPandey1 commented:
If user skip duplicate songs then only distinct songs are added in playlist 
![navidromeSkip](https://user-images.githubusercontent.com/57366926/111059901-f2f92900-84be-11eb-9900-cbe7a2a744bb.gif)

If user want to add duplicate song : 
![navidromeAdd](https://user-images.githubusercontent.com/57366926/111059934-3b184b80-84bf-11eb-8441-59a64cc32c3e.gif)

deluan commented on ui/src/i18n/en.json:
```diff
@@ -1,341 +1,346 @@
+      },
+      "message": {
+        "duplicate_song": "Add duplicate song ",
+        "song_exist": "There are duplicates being added to the playlist. Would you like to add the duplicates or skip them ?"
```
Small nitpick: Can you please remove the space before the question mark?

RitikPandey1 replied:
I have updated the message

deluan reviewed (commented):
This looks great. Besides the small nitpick below, I think we can improve the "Skip" behaviour: Instead of returning to the `AddToPlaylist` dialog when clicking the "Skip" button, it should just go ahead and add the non-duplicated songs to the playlist, closing all dialogs, similar to when you press the "Add" button.

RitikPandey1 commented:
> This looks great. Besides the small nitpick below, I think we can improve the "Skip" behaviour: Instead of returning to the `AddToPlaylist` dialog when clicking the "Skip" button, it should just go ahead and add the non-duplicated songs to the playlist, closing all dialogs, similar to when you press the "Add" button.
> 
> Also, if there were no changes to the Go codebase, there should not be any changes to the `go.sum` file. Can you please rebase and remove it from the commit?

All the mentioned changes are implemented but I'm not able to remove the go.sum file from my previous commit, please tell me  how can I remove go.sum file ?

deluan commented:
Take a look at `git rebase -i` (rebase interactively). If you can't do it, let me know and I can fix this, but this is definitely something you should learn, very helpful.

deluan commented on ui/src/i18n/en.json:
```diff
@@ -1,341 +1,346 @@
+        "export": "Export"
+      },
+      "message": {
+        "duplicate_song": "Add duplicate song ",
```
This should be `"Add duplicated songs`

RitikPandey1 replied:
I have changed the message

RitikPandey1 commented:
I just noticed that build was not succesful and showing error ` missing go.sum entry`, Is this because I removed go.sum file from my previous commit? I'm little confuse beacuse I didn't touch go.sum file when I was making initial changes and when I removed it then build wast not succesful, could you please tell me what I did wrong and how can I resolve this issue ? @deluan

jvoisin commented:
You have to keep the go.sum and go.mod files, without modifying them. maybe you removed them from your filesystem while removing them from your commit?

RitikPandey1 commented:
> You have to keep the go.sum and go.mod files, without modifying them. maybe you removed them from your filesystem while removing them from your commit?

No these files are in my filesystem

RitikPandey1 commented:
I have added unmodified go.sum file

RitikPandey1 commented:
Requesting review
@deluan

deluan commented:
Found a bug. I had a playlist with one song. If I tried to add 3 songs (including the one that's already there) and press Skip, it worked as expected, only adding the other two. But if I try to add again the same 3 songs (now all of them are in the playlist), and press Skip again, it adds all three again!
![Screen Recording 2021-03-17 at 12 24 37 AM](https://user-images.githubusercontent.com/331353/111415105-a4fb4580-86b7-11eb-9dce-fbc4e6b32a65.gif)

RitikPandey1 commented:
I'll try to debug this.

RitikPandey1 commented:
Bug fixed.
![navidromeskipsong](https://user-images.githubusercontent.com/57366926/111421434-fc7acf00-8712-11eb-9020-72bf700c8944.gif)

RitikPandey1 commented:
Requesting review 
@deluan

deluan commented:
Thanks for the PR!
